lib/tools/system-calls.js

```diff
@@ -771,6 +772,21 @@ systemCallMethods.waitForDevice = async function (appDeviceReadyTimeout = 30) {
   });
 };
 
+systemCallMethods.privilegedShell = async function (args) {
+  try {
+    await this.shell(args);
+  } catch (err) {
+    if (err.message.includes('must be root') === -1) {
+      throw err;
+    }
+    // this device needs adb to be running as root to run the specific command.
+    // so try to restart the daemon
+    log.debug(`Device requires adb to be running as root in order to run "adb shell ${args.join(",")}". Restarting daemon`);
```

Comment:
you could use shell-quote lib in order to properly print args

Reply:
```javascript
import { quote } from 'shell-quote';
...

`"adb shell ${quote(args)}". Restarting daemon`
```